pkg/controllers/capr/bootstrap/controller.go

```diff
@@ -530,16 +540,24 @@ func (h *handler) reconcileMachinePreTerminateAnnotation(bootstrap *rkev1.RKEBoo
 		h.rkeBootstrap.EnqueueAfter(bootstrap.Namespace, bootstrap.Name, 5*time.Second)
 		return bootstrap, generic.ErrSkip
 	}
-	return h.ensureMachinePreTerminateAnnotationRemoved(bootstrap, machine)
+
+	return h.ensureMachinePreDrainAnnotationRemoved(bootstrap, machine)
 }
 
-// ensureMachinePreTerminateAnnotationRemoved removes the pre-terminate annotation from a CAPI machine when we removing the rkebootstrap, indicating the infrastructure can be deleted.
-func (h *handler) ensureMachinePreTerminateAnnotationRemoved(bootstrap *rkev1.RKEBootstrap, machine *capi.Machine) (*rkev1.RKEBootstrap, error) {
+// ensureMachinePreDrainAnnotationRemoved removes the pre-drain annotation from a CAPI machine when we remove
+// the rkebootstrap, indicating the node can be drained. This also removes the legacy capiMachinePreTerminateAnnotation
+// if it exists.
+func (h *handler) ensureMachinePreDrainAnnotationRemoved(bootstrap *rkev1.RKEBootstrap, machine *capi.Machine) (*rkev1.RKEBootstrap, error) {
 	if machine == nil || machine.Annotations == nil {
 		return bootstrap, nil
 	}
 
 	var err error
+	if _, ok := machine.GetAnnotations()[capiMachinePreDrainAnnotation]; ok {
+		machine = machine.DeepCopy()
+		delete(machine.Annotations, capiMachinePreDrainAnnotation)
+		_, err = h.machineClient.Update(machine)
```

Comment:
If an error occurs here and then again below, this error will be swallowed. Is that OK or do we also need to add something like `if err != nil` here?

Reply:
Very good catch